src/coreclr/jit/morph.cpp

```diff
@@ -6384,11 +6384,6 @@ void Compiler::fgValidateIRForTailCall(GenTreeCall* call)
                 return WALK_ABORT;
             }
 
-            // GT_NOP might appear due to stores that end up as
-            // self-stores, which get morphed to GT_NOP.
-            if (tree->OperIs(GT_NOP))
-            {
-            }
```

Comment:
I don't think this one can be removed. Block morphing will still morph these to `GT_NOP`.

Reply:
but um.. isn't a no-op code? it checks for GT_NOP and does nothing?

Reply:
ah, I see! didn't notice `else`

Reply:
Feel free to add a `// fallthrough` or something to make it clear 🙂